java/arcs/core/allocator/Arc.kt

```diff
@@ -97,21 +91,29 @@ class Arc internal constructor(
     }
 
     /** Called whenever the [ArcState] changes to [Running]. */
-    fun onRunning(handler: () -> Unit) = onArcStateChangeFiltered(Running, handler)
+    fun onRunning(handler: () -> Unit) = onArcStateChangeFiltered(ArcState.Running, handler)
 
     /** Called whenever the [ArcState] changes to [Stopped]. */
-    fun onStopped(handler: () -> Unit) = onArcStateChangeFiltered(Stopped, handler)
+    fun onStopped(handler: () -> Unit) = onArcStateChangeFiltered(ArcState.Stopped, handler)
 
     /** Called whenever the [ArcState] changes to [Error]. */
-    fun onError(handler: () -> Unit) = onArcStateChangeFiltered(Error, handler)
-
-    private fun recomputeArcState(states: Collection<ArcState>): ArcState = when {
-        states.any { it == Deleted } -> Deleted
-        states.any { it == Error } -> Error
-        states.all { it == Running } -> Running
-        states.all { it == Stopped } -> Stopped
-        states.all { it == NeverStarted } -> NeverStarted
-        else -> Indeterminate
+    fun onError(handler: () -> Unit) = onArcStateChangeFiltered(ArcState.Error, handler)
+
+    private fun recomputeArcState(states: Collection<ArcState>): ArcState {
+        var commonState = ArcState.Indeterminate
+        for (state in states) {
```

Comment:
This is optional, since I'm ok with it, but some Kotlin readability reviewers tend to prefer states.forEach { state -> } instead of for (state in states) { ... }

Reply:
Ah yes, functional is the new hotness. Done.